synapse/handlers/room_member.py

```diff
@@ -218,6 +220,52 @@ def _local_membership_update(
                 newly_joined = prev_member_event.membership != Membership.JOIN
             if newly_joined:
                 yield self._user_joined_room(target, room_id)
+
+            # Copy over direct message status and room tags if this is a join
+            # on an upgraded room
+
+            # Check if this is an upgraded room
+            state_ids = yield self.store.get_current_state_ids(room_id)
+            create_id = state_ids.get((EventTypes.Create, ""))
+            if not create_id:
+                return
+            create_event = yield self.store.get_event(create_id)
+
+            if "predecessor" in create_event["content"]:
+                old_room_id = create_event["content"]["predecessor"]["room_id"]
+
+                # Retrieve room account data for predecessor room
+                user_account_data = yield self.store.get_account_data_for_user(
```

Comment:
given you just want the first item in the tuple you can write:

```
user_account_data, _ = yield ...
```

which then saves the slightly cryptic `[0]` indexes later

Reply:
Ohh, nice one. I love destructuring.